spirv_msl.cpp

```diff
@@ -10664,6 +11615,42 @@ bool CompilerMSL::OpCodePreprocessor::handle(Op opcode, const uint32_t *args, ui
 		break;
 	}
 
+	// Fix tessellation patch function processing
+	case OpLoad:
+	{
+		if (compiler.get_execution_model() == ExecutionModelTessellationControl)
+		{
+			uint32_t id = args[1];
+			uint32_t ptr = args[2];
+
+			uint32_t source_id = ptr;
+			auto *var = compiler.maybe_get_backing_variable(source_id);
+			if (var)
+				source_id = var->self;
+
+			// Only interested in standalone builtin variables.
+			if (compiler.has_decoration(source_id, DecorationBuiltIn))
+			{
+				auto builtin = static_cast<BuiltIn>(compiler.get_decoration(source_id, DecorationBuiltIn));
+				switch (builtin)
+				{
+				case BuiltInInvocationId:
+					invocation_ids[id] = ptr;
+					break;
+				default:
+					break;
+				}
+			}
+		}
+		break;
+	}
+
+	case OpControlBarrier:
+	{
+		passed_control_barrier = true;
```

Comment:
Trying to understand this part of the PR now ...
I believe this one is trying to detect when the shader has entered its "patch constant" phase.